opentelemetry/src/sdk/trace/sampler.rs

```diff
@@ -116,7 +118,8 @@ impl ShouldSample for Sampler {
                     .should_sample(parent_context, trace_id, name, span_kind, attributes, links)
                     .decision,
                 |ctx| {
-                    if ctx.is_sampled() {
+                    let parent_span_context = ctx.span().span_context();
+                    if parent_span_context.is_valid() && parent_span_context.is_sampled() {
```

Comment:
> The Span's SpanContext may be invalid to indicate a root span

I think we should allow invalid span context here

Reply:
You're right. I re-read the Built-in Samplers specifications and that seems implied. Thanks.